### PR TITLE
Remove handling of the Markdown component

### PR DESCRIPTION
**Pull request by Princesseuh**
## Changes

**Wait! This should only get merged once the core repo is ready for it!** ⚠️

The Markdown component is now deprecated in Astro, so we don't need to handle it anymore, this is the Prettier plugin version of all the PRs around the Astro repos removing handling of the Markdown component

Fix https://github.com/withastro/prettier-plugin-astro/issues/232

## Testing

Removed tests for it

## Docs

N/A

### CODE REVIEW

File: test/fixtures/markdown/embedded-in-markdown/input.md [hunk 1]
```diff
@@ -1,332 +0,0 @@
...
-Astro component syntax is a superset of HTML. **If you know HTML, you already know enough to write your first Astro component.**
-
-For example, this three-line file is a valid Astro component:
-
-```astro
-<!-- Example1.astro - Static HTML is a valid Astro component! -->
-  <div class="example-1">
-  <h1>Hello world!</h1>
-</div>
-```
```
Comment (antonyfaris) [inline test/fixtures/markdown/embedded-in-markdown/input.md:27]: This is not related to the markdown component. This means that inside markdown files you could format Astro syntax. I think we should support this.

File: test/tests/markdown.test.ts [hunk 2]
```diff
@@ -1,29 +0,0 @@
...
-	assert: { type: 'raw' },
-});
-
-// *** MARKDOWN ***
-// test(
-//   'can format an Astro file containing an Astro file embedded in a codeblock',
-//   files,
-//   'markdown/embedded-in-markdown',
-//   { mode: 'markdown' }
-// );
```
Comment (antonyfaris) [inline test/tests/markdown.test.ts:13]: This was commented out due to another bug in the formatting. I believe we should support this.

File: test/fixtures/markdown/embedded-in-markdown/input.md (hunk 1 above)
Comment (Princesseuh) [reply test/fixtures/markdown/embedded-in-markdown/input.md:27]: Oops, thought everything in Markdown was related to the Markdown component. Will restore those!
